package.json

```
@@ -0,0 +1,69 @@
+{
+  "name": "team-g-project-skeleton",
+  "version": "0.0.1",
+  "description": "",
+  "author": "",
+  "private": true,
+  "license": "UNLICENSED",
+  "scripts": {
+    "build": "nest build",
+    "format": "prettier --write \"src/**/*.ts\" \"test/**/*.ts\"",
+    "start": "nest start",
+    "start:dev": "nest start --watch",
+    "start:debug": "nest start --debug --watch",
+    "start:prod": "node dist/main",
+    "lint": "eslint \"{src,apps,libs,test}/**/*.ts\" --fix",
+    "test": "jest",
+    "test:watch": "jest --watch",
+    "test:cov": "jest --coverage",
+    "test:debug": "node --inspect-brk -r tsconfig-paths/register -r ts-node/register node_modules/.bin/jest --runInBand",
+    "test:e2e": "jest --config ./test/jest-e2e.json"
+  },
+  "dependencies": {
+    "@nestjs/common": "^10.0.0",
+    "@nestjs/core": "^10.0.0",
+    "@nestjs/platform-express": "^10.0.0",
+    "reflect-metadata": "^0.1.13",
+    "rxjs": "^7.8.1"
+  },
+  "devDependencies": {
+    "@nestjs/cli": "^10.0.0",
+    "@nestjs/schematics": "^10.0.0",
+    "@nestjs/testing": "^10.0.0",
+    "@types/express": "^4.17.17",
+    "@types/jest": "^29.5.2",
+    "@types/node": "^20.3.1",
+    "@types/supertest": "^2.0.12",
+    "@typescript-eslint/eslint-plugin": "^6.0.0",
+    "@typescript-eslint/parser": "^6.0.0",
+    "eslint": "^8.42.0",
+    "eslint-config-prettier": "^9.0.0",
+    "eslint-plugin-prettier": "^5.0.0",
+    "jest": "^29.5.0",
+    "prettier": "^3.0.0",
+    "source-map-support": "^0.5.21",
+    "supertest": "^6.3.3",
+    "ts-jest": "^29.1.0",
+    "ts-loader": "^9.4.3",
+    "ts-node": "^10.9.1",
+    "tsconfig-paths": "^4.2.0",
+    "typescript": "^5.1.3"
+  },
```

```suggestion
{"packageManager": "yarn@1.22.19",
	"engines": {
		"node": "18.16.1"
	},
```

제가 전에 개인프로젝트 할 때 노드버전과 패키지매니저 버전을 고정하려고 사용했었는데요.
버전 안정성을 위해 노드 버전을 명확히 지정하거나, 몇 버전 이상의 버전을 사용해야 한다는 등 규칙을 적을 수 있습니다.
docker 이미지가 돌아가는 환경의 노드버전과 일치시키면 어떨까 싶습니다.(가능하다면 패키지 매니저버전까지)

저같은 경우는 node 18.16 버전을 사용중인데, docker에 설정하신 노드버전을 보면 20이더라구요.
다른 분들도 노드버전이 다르시지 않을까 싶어, 명시해두면 좋을 것 같아 제안드립니다.

참고로 로컬에서 node 버전은 매번 바꿔주기 귀찮을 수 있으니 프로젝트의 루트에 .nvmrc 파일 추가하고 몇 가지 쉘 스크립트를 작성하면 패키지별로 지정한 노드버전이 다르더라도 해당 버전으로 작업이 됩니다.
(참고: https://hyeok999.github.io/2020/06/02/NVM/#a5)

---

> 저같은 경우는 node 18.16 버전을 사용중인데, docker에 설정하신 노드버전을 보면 20이더라구요.
> 다른 분들도 노드버전이 다르시지 않을까 싶어, 명시해두면 좋을 것 같아 제안드립니다.

이 부분에서 제 의도와는 다른거 같아서 코멘트 남깁니다. 제가 docker compose를 사용한 부분은 로컬 환경에서 노드 버전이 모두 다르기 때문에 `docker compose up`을 사용하게 되면 node 20 버전으로 도커 환경 실행되니까 모두 같은 노드 버전의 환경에서 작업해볼 수 있기 때문에 도커를 사용했는데요.
로컬에서까지 노드 버전을 맞춰서 작업하게되면 도커 컴포즈가 의미가 없게 되는거 같습니다.

---

아 네넵 그부분은 이해했습니다.

그럼 도커 컨테이너 위에서 의존성 패키지를 npm install 하게되면 도커 컨테이너의 node version에 맞게 의존성 버전이 제대로 설치된다는 말씀이시죠?

저는 그 부분까지는 몰랐고, 도커를 사용하지 않으면 npm install은 로컬의 node version을 따라가니까 일치해야하지 않나 생각했었어요😅

그렇다면, "꼭" 도커 컨테이너를 띄우고 작업해야겠군요!